### PR TITLE
DOC: add note about isinstance for frozen rvs

### PR DESCRIPTION
**Pull request by stsievert**
This addresses #8807

### CODE REVIEW

**Comment by stsievert** [comment]
Changes made. Thanks for those, this is more useful and what I was looking for.

**Review by andyfaff — COMMENTED** [review]
Hmm, still 50:50 on this.

How about line 1443 gets changed to `rv_frozen object with...`

**Comment by stsievert** [comment]
Like that?

File: scipy/stats/_distn_infrastructure.py [hunk 1]
```diff
@@ -1440,9 +1440,9 @@ class rv_continuous(rv_generic):
     location, and scale parameters returning a "frozen" continuous RV object:
 
     rv = generic(<shape(s)>, loc=0, scale=1)
-        frozen RV object with the same methods but holding the given shape,
+        stats.distributions.rv_frozen object with the same methods but holding the given shape,
```
Comment (andyfaff) [inline scipy/stats/_distn_infrastructure.py]: Better, I was thinking of:

```
`rv_frozen` object with
```

I _think_ the class name needs to go in backticks.

Comment (stsievert) [reply scipy/stats/_distn_infrastructure.py]: I think you're right.

At line 1316 there's

>     `rv_continuous` is a base class ...

​In HTML we see

> [rv_continuous][1] is a base class ...

[1]:https://docs.scipy.org/doc/scipy/reference/generated/scipy.stats.rv_continuous.html#scipy.stats.rv_continuous

**Pull request merged**